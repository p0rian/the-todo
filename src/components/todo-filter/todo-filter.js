import { ToggleButton, ToggleButtonGroup, Box } from "@mui/material";
import React from "react";
import PropTypes from "prop-types";

export const TodoFilter = ({ filter, handleFilterChange }) => (
  <Box sx={{ display: "flex", justifyContent: "center", mb: 2 }}>
    <ToggleButtonGroup
      value={filter}
      exclusive
      onChange={handleFilterChange}
      aria-label='todo filter'
    >
      <ToggleButton value='all' aria-label='show all todos'>
        All
      </ToggleButton>
      <ToggleButton value='active' aria-label='show active todos'>
        Active
      </ToggleButton>
      <ToggleButton value='completed' aria-label='show completed todos'>
        Completed
      </ToggleButton>
    </ToggleButtonGroup>
  </Box>
);

TodoFilter.propTypes = {
  filter: PropTypes.string.isRequired,
  handleFilterChange: PropTypes.func.isRequired,
};
