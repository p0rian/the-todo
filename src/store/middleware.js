import { createListenerMiddleware, isAnyOf } from "@reduxjs/toolkit";
import { addTodo, updateTodo, removeTodo } from "./slices/todoSlice";

export const listenerMiddleware = createListenerMiddleware();
listenerMiddleware.startListening({
  matcher: isAnyOf(addTodo, updateTodo, removeTodo),
  effect: (_, listenerApi) => {
    localStorage.setItem("todoData", JSON.stringify(listenerApi.getState()));
  },
});
