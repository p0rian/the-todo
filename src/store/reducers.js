import { todoReducer } from "./slices/todoSlice";

export const reducers = {
  todoData: todoReducer,
};
