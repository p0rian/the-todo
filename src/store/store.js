import { configureStore } from "@reduxjs/toolkit";
import { reducers } from "./reducers";
import { listenerMiddleware } from "./middleware";

const todoState = JSON.parse(localStorage.getItem("todoData") || "null");
const preloadedState = todoState ? { todoData: todoState.todoData } : {};

export const store = configureStore({
  preloadedState,
  reducer: { ...reducers },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(listenerMiddleware.middleware),
});
