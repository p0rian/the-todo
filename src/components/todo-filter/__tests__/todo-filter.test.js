import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { store } from "store/store";
import { TodoFilter } from "../todo-filter";

describe("TodoFilter component", () => {
  it("renders correctly", () => {
    const filter = "all";
    const handleFilterChange = jest.fn();
    const { container } = render(
      <Provider store={store}>
        <TodoFilter filter={filter} handleFilterChange={handleFilterChange} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
