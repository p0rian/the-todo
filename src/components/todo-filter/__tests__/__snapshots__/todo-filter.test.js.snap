// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilter component renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1og1qar"
  >
    <div
      aria-label="todo filter"
      class="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
      role="group"
    >
      <button
        aria-label="show all todos"
        aria-pressed="true"
        class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="all"
      >
        All
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="show active todos"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="active"
      >
        Active
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="show completed todos"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="completed"
      >
        Completed
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
